docs/tutorials/HD143006_Part_1.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# + nbsphinx="hidden"
+# %matplotlib inline
+
+# + nbsphinx="hidden"
+# %run notebook_setup
+# -
+
+# # HD143006 Tutorial Part 1
+# 
+# Here we are going to do the first step in real world data analysis with MPoL. We will gather the data and then produce a diagnostic image-- the MPoL Dirty Image. This image will have no RML methods applied to it, but it is still a good preliminary checkpoint. We will see the quality of this image and compare it to the DSHARP survey's image ([Andrews et al. 2018](https://ui.adsabs.harvard.edu/abs/2018ApJ...869L..41A/abstract)) created using the CLEAN method.
+# 
+# ### Introduction to Regularized Maximum Likelihood (RML) Imaging
+# 
+# Regularized Maximum Likelihood (RML) imaging is a forward modeling methodology. We predict an image then represent it as an array of pixels. This is brought to the visibility domain through a forward Fourier transform. This is then compared to the measured data to ensure it is a viable predicted image. Due to information loss of the true image when obtaining the measured data, several predicted images- including incorrect ones- will match. To get to our best predicted image, we make new predictions by choosing the most likely (Maximum Likelihood) configuration and favoring specified criteria (Regularized). These criteria or regularizers, are chosen by the user. Some examples of favored criteria are smoothness and sparsity. The likeliness and how well a predicted image meet a certain criterion is mathematically represented in a loss function that contains hyperparameters used to weight data and regularizers. We minimize this loss function by performing a gradient descent, in which we adjust the hyperparameters. When the loss function is minimized, our predicted image is at its best version to fit the collected data and follow our specified criterion.
```

Comment:
After some discussion we determined that this may not be correct in that hyperparameters are set by the user (unless using something such as Ray Tune) and parameters are what are adjusted during gradient descent. Though I think there is still some grey area with these parameters. Would it be pixel size and/or Jy/arcsec^2 within each section of the cube that is changed with the gradient descent? If it is, how would this not result in the gradient descent producing our "true image" that we are comparing it with? Would it be because of the loss function not only looking at a comparison of the data from the true image but also taking into account what is most likely with the option to further analyze with regularizers?

Reply:
This is a good question, and the answer is kind of subtle and kind of meta. The short answer is that yes, pixel size and number of pixels are technically parameters of the model, just like the way the intensity values of the pixels are parameters. But, in general, we'll want to leave pixel size and number of pixels constant during the fit.

The long answer is that this boils down to how one represents information (like an image). By choosing some pixel sampling rate, we're essentially asserting that the image is "band-limited" to those spatial frequencies. More information [here](https://en.wikipedia.org/wiki/Nyquist%E2%80%93Shannon_sampling_theorem).

Reply:
@iancze Thank you! That helps me understand why we don't get the true image. I'm still a little unclear as to the adjustments made in the gradient descent. Above we state during the gradient descent the hyperparameters are what are being changed- or in the wording of Brianna's paper the coefficients corresponding to each regularizer. After reading Brianna's paper I'm wondering if it would instead be the pixel values usually represented with units of Jy arcsec^-2  that are being changed.

Reply:
Perhaps I'm not quite following what you mean by "true image" in this context. In the case of HD 143006, we don't know what truth is, since this is real data! If we were to make a mock simulated dataset from a "true" sky image, like in the case of the ALMA logo, then we can go back and reference truth in that case.

The term "hyperparameters" can be thrown around a little loosely sometimes. First, hyperparameters *are* technically parameters, but the reason why they are "hyper" is because they are one level meta than the parameters under investigation in the base scenario. In the example of an RML fit, usually we would call the pixel value intensities (in Jy arcsec^2, or any unit, really) the parameters of the model. If we were doing a simple line fit to some x,y data, for example, we would call the slope and intercept the parameters of the model. It's these base level parameters that are what's being optimized by gradient descent.

We would typically call things like "regularization strength" or the lambda terms hyperparameters. For any given optimization run, we'll usually run with these parameters fixed. But there is the question of whether the hyperparameters could be varied to produce a "better" image (in the case of real data, better as measured by cross-validaton scroe). So this is where things like hyperparameter tuning (with Ray Tune) come in. There's a "meta-level" to the optimization, in that we're trying optimization with a bunch of different hyperparameter settings.

We could also treat the pixel sizes and number of pixels like hyperparameters, since they will have a (hopefully small) effect on the image that results. For now we're just ignoring this effect in the hyperparameter training to save computational effort.

Reply:
I think you covered my questions and provided some helpful information! In a nutshell, I was thinking change "We minimize this loss function by performing a gradient descent, in which we adjust the hyperparameters," to "We minimize this loss function by performing a gradient descent, in which we adjust the pixel value intensities. Within this optimization run, hyperparameters are usually held fixed, but can be tuned between runs to produce a better image," though I wasn't sure if this was correct. However this conversation cleared it up! Thank you for your help @iancze !

Reply:
Right, that's correct! The hyperparameters will not typically be changed during the optimization run or by gradient descent. I'm working on a few changes to the developer documentation + small updates to docs, so I can make this change on that branch, which I'll push and merge in a second.

Reply:
Great! Thanks!